godot-kotlin/godot-library/src/nativeMain/kotlin/godot/annotation/PropertyTypeHintAnnotation.kt

```diff
@@ -0,0 +1,246 @@
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Layers3DRender
+
+/**
+ * Can only be used on Int properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ObjectId
+
+/**
+ * Can only be used on Int properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class IntRange(val start: Int, val end: Int, val step: Int = -1, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on Float properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class FloatRange(val start: Float, val end: Float, val step: Float = -1f, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class DoubleRange(val start: Double, val end: Double, val step: Double = -1.0, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ExpEasing(val attenuation: Boolean = false, val inout: Boolean = true)
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ExpRange(val start: Float, val end: Float, val step: Float = -1f, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class MultilineText
+
+/**
+ * This property hint is ignored! It's here for completion sake and match parity with godot's export hints.
+ *
+ * A String that is visible to the editor is always marked as PROPERTY_HINT_TYPE_STRING if not defined otherwise
+ * through another PropertyTypeHintAnnotation
+ *
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class StringTypeHint
+
+/**
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Dir
+
+/**
+ * Can only be used on String and File properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class File(vararg val extensions: String = [])
+
+/**
+ * Can only be used on String and File properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class SaveFile(vararg val extensions: String = [])
+
+/**
+ * Can only be used on String and File properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class GlobalDir
+
+/**
+ * Can only be used on String and File properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class GlobalFile(vararg val extensions: String = [])
```

Comment:
same here, drop in favor of `File` with global property.

Reply:
:+1: